trio/_sync.py

```diff
@@ -970,8 +970,9 @@ def abort_fn(_):
         self._get_wait[task] = None
         try:
             value = await _core.wait_task_rescheduled(abort_fn)
-        finally:
+        except BaseException:
             self._get_wait.pop(task, None)
```

Comment:
As stated in the discussion in #553, I think the place for this cleanup is inside `abort_fn` and we don't need a try block here.